rust/agama-server/src/software/web.rs

```diff
@@ -99,6 +118,57 @@ async fn patterns_changed_stream(
     Ok(stream)
 }
 
+async fn registration_requirement_changed_stream(dbus: zbus::Connection,
+) -> Result<impl Stream<Item = Event>, Error> {
+    let proxy = RegistrationProxy::new(&dbus).await?;
+    let stream = proxy
+        .receive_requirement_changed()
+        .await
+        .then(|change| async move {
+            if let Ok(id) = change.get().await {
+                // unwrap is safe as possible numbers is send by our controlled dbus
+                return Some(Event::RegistrationRequirementChanged { requirement: id.try_into().unwrap() });
+            }
+            None
+        })
+        .filter_map(|e| e);
+    Ok(stream)
+}
+
+async fn registration_email_changed_stream(dbus: zbus::Connection,
+) -> Result<impl Stream<Item = Event>, Error> {
+    let proxy = RegistrationProxy::new(&dbus).await?;
+    let stream = proxy
+        .receive_email_changed()
+        .await
+        .then(|change| async move {
+            if let Ok(_id) = change.get().await {
+                // unwrap is safe as possible numbers is send by our controlled dbus
+                return Some(Event::RegistrationChanged);
```

Comment:
So we need to ask for the registration information again, right? Perhaps it would be better to send the registration information along with the event.

Reply:
well, it is possible, I will just need to recreate info again from proxy as I cannot get it from signal as it contain just one param and not all of them.

Reply:
We already solve a problem like this with the [ProgressStream](https://github.com/openSUSE/agama/blob/architecture_2024/rust/agama-server/src/web/common.rs#L173-L220) API, that includes the full struct in the events. We can folllow a similar approach here.

However, we decided to keep it simple by now, but we have plans to improve it later.